src/Nethermind/Nethermind.Serialization.Rlp/ReceiptRecoveryBlock.cs

```diff
@@ -51,9 +51,12 @@ public Transaction GetNextTransaction()
             return _transactions[_currentTransactionIndex++];
         }
 
-        Rlp.ValueDecoderContext decoderContext = new(_transactionData, true);
-        decoderContext.Position = _currentTransactionPosition;
-        TxDecoder.InstanceWithoutLazyHash.Decode(ref decoderContext, ref _txBuffer, RlpBehaviors.AllowUnsigned);
+        Rlp.ValueDecoderContext decoderContext = new(_transactionData, true)
+        {
+            Position = _currentTransactionPosition
+        };
+        TxDecoder.Instance.Decode(ref decoderContext, ref _txBuffer, RlpBehaviors.AllowUnsigned);
+        Hash256 _ = _txBuffer.Hash; // Force Hash evaluation
```

Comment:
This was the only instance where the hash was evaluated eagerly. Instead of distorting the API to conform to this single use case, we force the hash evaluation immediately.